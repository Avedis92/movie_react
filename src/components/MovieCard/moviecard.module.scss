@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.container {
  background-color: white;
  width: 100%;
  .movieItem {
    height: 80%;
    img {
      width: 100%;
      height: 100%;
      cursor: pointer;
    }
    .movieInfo {
      padding-left: 5%;
      padding-top: 5%;
      @include flexElements(flex, row, center, space-between);
      height: max-content;
      .titleAndGenres {
        h3 {
          margin-bottom: 7%;
        }
        h4 {
          display: inline;
          color: rgb(27, 186, 235);
        }
      }
      .votesAverage {
        border: 2px solid rgb(27, 186, 235);
        border-radius: 5px;
        width: 20%;
        margin-right: 4%;
        text-align: center;
        color: rgb(27, 186, 235);
        p {
          margin-top: 2%;
          margin-bottom: 2%;
          font-size: 18px;
        }
      }
    }
  }
}
