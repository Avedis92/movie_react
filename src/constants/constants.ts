export const apiKey = '0600befd930f4be78f3d1123f393affc';
export const baseUrl = 'https://api.themoviedb.org/3/';
export const trendingMovieUrl = `${baseUrl}trending/movie/week?api_key=${apiKey}&page=`;
export const topRatedMovieUrl = `${baseUrl}movie/top_rated?api_key=${apiKey}&page=`;
export const upcomingMovieUrl = `${baseUrl}movie/upcoming?api_key=${apiKey}&page=`;
export const movieGenreListUrl = `${baseUrl}genre/movie/list?api_key=${apiKey}`;
export const searchMovieUrl = `${baseUrl}search/movie?api_key=${apiKey}&query=`;
export const postPathUrl = 'https://image.tmdb.org/t/p/w500';
export const defaultBackgroundImage = 'https://media-assets-05.thedrum.com/cache/images/thedrum-prod/s3-news-tmp-10557-netflixa--default--1280.jpg';
export const pageNotFoundBackgroundImage = 'https://cdn3.vectorstock.com/i/1000x1000/67/12/404-error-page-background-vector-20596712.jpg';
export const movieVideoUrl = `${baseUrl}/movie/`;
export const searchMovieByGenreUrl = `${baseUrl}discover/movie?api_key=${apiKey}&with_genres=`;
export const defaultPageNumber = 1;
export const defaultMovieType = 'trendings';
