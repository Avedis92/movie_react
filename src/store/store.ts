import { configureStore } from '@reduxjs/toolkit';
import genreReducer from './slices/genre';

export const store = configureStore({
  reducer: {
    genre: genreReducer,
  },
});

export type RootState=ReturnType<typeof store.getState>;
export type MovieDispatch=typeof store.dispatch;
