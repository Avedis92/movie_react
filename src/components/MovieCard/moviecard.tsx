import { useState } from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';
import styles from './moviecard.module.scss';
import MovieCardMouseOver from '../MovieCardMouseOver/moviecardmouseover';
import ModalOpener from '../ModalOpener/modalopener';
import * as constants from '../../constants/constants';

function MovieCard({
  id, genreIds, title, voteAverage, overview, posterPath,
}:Card) {
  const allgenre = useSelector((state:RootState) => state.genre);
  const [movieCardVisibility, setMovieCardVisibility] = useState(true);
  const [viewInfoIsClicked, setViewInfoIsClicked] = useState(false);
  const movieGenre:Array<Genre> = allgenre.entities.length > 0 ? genreIds.map(
    (myid:number) => allgenre.entities.find((genre:Genre) => genre.id === myid),
  ) : [];
  const {
    container, movieItem, movieInfo, titleAndGenres, votesAverage,
  } = styles;
  return (

    <div className={container} onMouseLeave={() => setMovieCardVisibility(true)}>
      {!viewInfoIsClicked
        ? (
          <div className={movieItem}>
            {movieCardVisibility ? (
              <img
                src={posterPath !== `${constants.postPathUrl}null` ? posterPath : constants.defaultBackgroundImage}
                alt={title}
                onMouseOver={() => setMovieCardVisibility(false)}
              />
            )
              : (
                <ModalOpener
                  handleViewInfoClick={() => setViewInfoIsClicked(true)}
                  backgroundImage={posterPath}
                  movieId={id}
                />
              )}
            <div className={movieInfo}>
              <div className={titleAndGenres}>
                <h3 data-testid="movieTitle">{title}</h3>
                {allgenre.entities.length > 0 ? movieGenre.map((eachGenre:Genre) => (
                  <h4 key={eachGenre.id}>
                    {eachGenre.name}
                    {' '}
                  </h4>
                )) : null}
              </div>
              <div className={votesAverage}>
                <p>{voteAverage}</p>
              </div>
            </div>
          </div>
        )
        : (
          <MovieCardMouseOver
            key={id}
            id={id}
            title={title}
            voteAverage={voteAverage}
            overview={overview}
            posterPath={posterPath !== `${constants.postPathUrl}null` ? posterPath : constants.defaultBackgroundImage}
            handleClicks={() => {
              setViewInfoIsClicked(false);
              setMovieCardVisibility(true);
            }}
          />
        )}
    </div>
  );
}
export default MovieCard;
