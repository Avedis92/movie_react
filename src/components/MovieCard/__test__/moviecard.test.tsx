import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import MovieCard from '../moviecard';
import * as constants from '../../../constants/constants';

describe('Make tests on MovieCard component', () => {
  test('Make a snapshot test', () => {
    const {asFragment} = render(
      <Provider store={store}>
        <MovieCard
          id={297761}
          genreIds={[14, 28, 80]}
          title="Suicide Squad"
          voteAverage={5.91}
          overview="From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences."
          posterPath={`${constants.postPathUrl}/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg`}
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
