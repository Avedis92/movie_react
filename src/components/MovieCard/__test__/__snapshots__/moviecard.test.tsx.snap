// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make tests on MovieCard component Make a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="movieItem"
    >
      <img
        alt="Suicide Squad"
        src="https://image.tmdb.org/t/p/w500/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg"
      />
      <div
        class="movieInfo"
      >
        <div
          class="titleAndGenres"
        >
          <h3
            data-testid="movieTitle"
          >
            Suicide Squad
          </h3>
        </div>
        <div
          class="votesAverage"
        >
          <p>
            5.91
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
